File: content/atlas/source/includes/fts/field-types/embeddedDocuments/create-index-specified.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

database = client["sample_training"]
collection = database["companies"]

# Create the MongoDB Search index definition for the embeddedDocuments field with specified fields
search_index_model = SearchIndexModel(
    definition={
        "mappings": {
            "fields": {
                "offices": {
                    "type": "embeddedDocuments",
                    "dynamic": False,
                    "fields": {
                        "country_code": {
                            "type": "string"
                        },
                        "state_code": {
                            "type": "string"
                        }
                    }
                }
            }
        },
    },
    name="default",
)

# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")

File: content/atlas/source/includes/fts/field-types/embeddedDocuments/create-index-specified.js
const { MongoClient } = require("mongodb");

// Replace the connection string with your MongoDB deployment's connection string
const uri =  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_training");
    const collection = database.collection("companies");

    // Create the MongoDB Search index definition for the embeddedDocuments field with specified fields
    const index = {
        name: "default",
        definition: {
            "mappings": {
                "fields": {
                    "offices": {
                        "type": "embeddedDocuments",
                        "dynamic": false,
                        "fields": {
                            "country_code": {
                                "type": "string"
                            },
                            "state_code": {
                                "type": "string"
                            }
                        }
                    }
                }
            }
        }
    }

    // Create the index
    const result = await collection.createSearchIndex(index);
    console.log("New index name: " + result);
  } finally {
    await client.close();
  }
}

run().catch(console.dir);
